src/routes/exercise/+page.server.ts

```diff
@@ -0,0 +1,178 @@
+import type { PageServerLoad, Actions } from './$types';
+import { setError, superValidate } from 'sveltekit-superforms';
+import { fail, redirect } from '@sveltejs/kit';
+import { zod } from 'sveltekit-superforms/adapters';
+import { formSchema } from './schema';
+import { fetchExerciseData } from '$lib/requestHandler';
+import { handleAuthenticatedRequest } from '$lib/requestHandler';
+import { setIDEBoilerPlate } from '$lib/boilerplate';
+
+
+
+const backendUrl = import.meta.env.VITE_BACKEND_URL;
+const apiVersion = import.meta.env.VITE_V1;
+
+export const load: PageServerLoad = async ({ cookies, url }) => {
+	const access_token: string = cookies.get('access_token') || '';
+	const refresh_token: string = cookies.get('refresh_token') || '';
+	const exerciseId = url.searchParams.get('exerciseid');
+	const sessionId = url.searchParams.get('seshid');
+
+	const response = await fetch(`${backendUrl}${apiVersion}/exercises/${exerciseId}`, {
+		method: 'GET',
+		headers: {
+			'Content-Type': 'application/json',
+			Authorization: `Bearer ${access_token}`	
+	}});
+
+	const responseBody = await response.text();
+	let jsonResponse;
+
+    if (responseBody) {
+        try {
+            jsonResponse = JSON.parse(responseBody);
+            delete jsonResponse.solution;
+			jsonResponse.testCases = jsonResponse.testCases.filter(testCase => testCase.publicVisible);
+       
+        } catch (error) {
+            console.error('Failed to parse JSON response:', error);
+            throw new Error('Failed to parse JSON response');
+        }
+    } else {
+        jsonResponse = {};
+    }
+
+    console.log(jsonResponse);
+
+	const testTemplate = {
+        parameters: {
+            input: jsonResponse.testCases[0].inputParams.map((value, index) => ({
+                type: jsonResponse.inputParameterType[index],
+                value
+            })),
+            output: jsonResponse.testCases[0].outputParams.map((value, index) => ({
+                type: jsonResponse.outputParamaterType[index],
+                value
+            }))
+        }
+    };
+
+    let form = await superValidate(zod(formSchema));
+    form.data.codeText = setIDEBoilerPlate(testTemplate);
+
+	return {
+		form,
+		exerciseData: jsonResponse
+	};
+};
+
+function convertFormData(formData, sessionId) {
+	return {
+		"solution": formData.codeText,
+		"sessionId": sessionId,
+	}
+}
+
+async function postSolution(
+	backendUrl: string,
+	api_version: string,
+	access_token: string,
+	apiData,
+	exerciseId: int
+): Promise<Response> {
+	return await fetch(`${backendUrl}${api_version}/exercises/${exerciseId}/submission`, {
```

Comment:
We need some refreshing on these fetches

Reply:
Would require a new method as the request handler doesn't seem to work for anon tokens?